### ISSUE TITLE
Feature Request: Slider to define quantity to purchase

### ISSUE THREAD

**Issue by anubisthejackle** [doublespeakgames/adarkroom#118]
I have a lot of fur. No, really, a _lot_ of fur. I've been stocking it up for a while now to buy cheap scales and teeth when the wanderer showed up. But now that I have almost 40,000 fur, that's a lot of clicking. I'd love to be able to click a button (possibly just a simple +) to open a slider that let's me define a quantity to purchase of an item.

**Comment by Julienlego** [comment]
Sounds like a fun thing to do to contribute. @chuglydude  have you made any progress in doing this? cause if not I'd like to give it a try :)

**Comment by chuglydude** [comment]
What is this regarding

**Comment by chuglydude** [comment]
Oh yah I am working on this case still

**Comment by chuglydude** [comment]
sorry i can't do this right know but i will give you chance

**Comment by Julienlego** [comment]
hey @chuglydude what did you have in mind in where to put the slider? I thought of making a simply (+) button as an additional button for the nomad, but then I thought, why not be able to have a slider for all selling options.

**Comment by chuglydude** [comment]
Put it on the right

**Comment by trcollinson** [comment]
@Julienlego I'd just go for it! If you've got the means, put it in a PR

**Comment by Julienlego** [comment]
I found out that I'd have to use jquery-ui to make the slider. I'm thinking of making a pop-up slider for when you click a buy/sell button and can buy/sell more than one. I'd rather not modify the <div> that the button is incorporate the slider because that would be too messy.

**Comment by skj3gg** [comment]
> use jquery-ui to make the slider, not sure if that's ok

that's really NOT ok, but I'm sure we can think of something around this, so no need to worry. Here are some ideas to explore and see what might work best. 

1. The button itself is a slider, overlay basically another div the size of the button and allow the user the ability to drag the right side of that div in. The percentage the div's width is to the buttons div is the percentage to sell. Sounds really intriguing but extremely difficult.

1. Hack a input field to look like a button and allow the user to type in the amount they want, being sure the validate that they have the resources for that.

1. Write a custom range slider and find a place to elegantly put it. It's probably a hundred lines of code, and not fun to figure out, but it's doable.

**Comment by trcollinson** [comment]
@skj3gg I don't want to sound disrespectful, as this is just a serious question and not a criticism or commentary. But, why is using jquery NOT ok?

**Comment by skj3gg** [comment]
We do use jQuery. (SIDENOTE: though @Continuities and I both agree if it was worth the time to rewrite ADR without jQuery, we would) jQuery UI is a different beast in itself though and the addition of it would add a rather unnecessary and large dependency.

**Comment by Continuities** [comment]
Yeah, I'd rather not include more libraries if we can avoid it. Something
like a slider isn't too difficult to implement manually. I'd be happy to
whip something up if necessary.

**Comment by skj3gg** [comment]
*cough* gridland *cough* *cough*

**Comment by Continuities** [comment]
It'll get open-sourced soon, I promise!

**Comment by Julienlego** [comment]
I tried another idea where there's a range slider at the top of "buySection" that lets you choose how much you want to buy of a "thing", then click one of the "buy" buttons. problem is that the range slider doesn't move. is it because it's in an "<input>"?

**Comment by anubisthejackle** [comment]
With as wide-spread as HTML 5 is these days, I think a simple `<input type="number" />` could do the trick here. It doesn't really _have_ to be a slider.

**Comment by AndySky21** [comment]
I don't think the fact `<input type="number" />` is not used on ADR has anything to do with HTML5 diffusion actually. It's just a matter of consistency across the platform (input-number would prove useful for workers, path supplies and ship alike).

However, as I have a 90%-ready update with a bunch of changes (most code stuff issue, but something involving interface too), I can make something up for this, too.

Note for the staff (@Continuities and @groteworld): we can discuss each part of the update, but most of them are intertwined so please don't hate me too much if I don't branch it up.

**Comment by ttymed** [comment]
Hello,
So is this issue dead or taken care of?  If not I'd like to make the slider.

**Comment by AndySky21** [comment]
I have a project for this, but I suppose you'd be faster.

**Comment by Julienlego** [comment]
Go for it. Sorry, I've been busy recently.

**Comment by nekochan34** [comment]
So has this been finished? I would love to try and make this (and if not a slider has anyone thought of more buttons (+/- 10, +/- 100) depending on how much you have?

**Comment by gitupofftheflooranddosomework** [comment]
Is this something we still want to do? This can be easily done with range,css, and javscript, if it is ill add it to my todo
